cardano-testnet/src/Testnet/Blockfrost.hs

```diff
@@ -0,0 +1,268 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Werror=missing-fields #-}
+
+module Testnet.Blockfrost
+  ( BlockfrostParams
+  , blockfrostToGenesis
+  ) where
+
+import           Cardano.Ledger.BaseTypes (EpochInterval, Nonce, NonNegativeInterval,
+                  UnitInterval, ProtVer(..), Version)
+import           Cardano.Ledger.Coin (Coin)
+import           Cardano.Ledger.Core (PParams(..))
+import           Cardano.Ledger.Shelley.Genesis (ShelleyGenesis(..))
+import           Cardano.Ledger.Shelley.PParams (ShelleyPParams(..))
+import           Cardano.Ledger.Alonzo.Genesis (AlonzoGenesis(..))
+import           Cardano.Ledger.Alonzo.PParams (CoinPerWord)
+import           Cardano.Ledger.Conway.Genesis (ConwayGenesis(..))
+import           Cardano.Ledger.Conway.PParams (UpgradeConwayPParams(..),
+                  PoolVotingThresholds(..), DRepVotingThresholds(..))
+import           Cardano.Ledger.Plutus (CostModel, CostModels, ExUnits(..),
+                  Language(..), Prices(..))
+import qualified Cardano.Ledger.Plutus.CostModels as CostModels
+
+import           Control.Applicative ((<|>))
+import qualified Data.Aeson as Aeson
+import           Data.Aeson (FromJSON(..), (.:), withObject)
+import qualified Data.Aeson.Types as Aeson
+import           Data.Aeson.Types (parseFail)
```

Comment:
A nitpick, but you're using `Aeson.parseFail` and also as unqualified. It would be nice to unify those.